### PR TITLE
feat(collection): Add support for string items

### PR DESCRIPTION
**Pull request by NicholasBoll**
## Summary

Add support for string items in the collection system. This includes string items. `slugify` was added to support strings to be used as id and attributes in HTML. This can be used to support autocomplete where the items are most likely to be string.

## Release Category
Components

---

## Checklist

- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)

## Areas for Feedback? (optional)

- [x] Code
- [x] Documentation
- [x] Testing

### CODE REVIEW

File: modules/react/collection/lib/useCursorListModel.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import {assert, createModelHook, Generic} from '@workday/canvas-kit-react/common';
 
-import {useBaseListModel, Item, defaultGetId as getId} from './useBaseListModel';
```
Comment (NicholasBoll) [inline modules/react/collection/lib/useCursorListModel.tsx:4]: This was so embarrassing. It prevented non-default item objects from being used

File: modules/react/collection/lib/useListItemRegister.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import {
   createElemPropsHook,
   useLocalRef,
   useForkRef,
+  slugify,
```
Comment (alanbsmith) [inline modules/react/collection/lib/useListItemRegister.tsx:8]: 🐌 - 🐚

File: modules/react/menu/lib/MenuItem.tsx [hunk 3]
```diff
@@ -149,7 +149,7 @@ export const useMenuItem = composeHooks(
         // We only need to run when the ID has changed. If the static API is used, the first time
```
Comment (alanbsmith) [inline modules/react/menu/lib/MenuItem.tsx:149]:
```suggestion
        // We only need to run when the ID or cursorId has changed. If the static API is used, the first time
```

**Review by alanbsmith — APPROVED** [review]
👍🏼

**Pull request merged**